### PR TITLE
Add unit tests for AWS Datasource and items

### PR DESCRIPTION
**Pull request by recca5p**

### CODE REVIEW

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
+using Reveal.Sdk.Dom.Data;
+using Xunit;
+
+namespace Reveal.Sdk.Dom.Tests.Data.DataSourceItems
+{
+    public class AmazonAthenaDataSourceItemFixture
+    {
+        [Fact]
+        public void Constructor_WithDataSource_SetsTitleAndDataSource()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:9]: Should be `Constructor_SetsTitleAndDataSource_WithDataSource` or `Constructor_SetsTitleAndDataSource_AsProvided`

Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:9]: We are using the name format is MethodName_ExpectedBehavior_Condition. So I think it's worth to check if the names are following this format

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            // Arrange
+            string title = "Test Item";
+            var dataSource = new AmazonAthenaDataSource();
+
+            // Act
+            var item = new AmazonAthenaDataSourceItem(title, dataSource);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:16]: Initializing an object instance should be in Arrange. Act only includes the method calling / state updating.

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+            // Assert
+            Assert.Equal(title, item.Title);
+            Assert.Equal(dataSource, item.DataSource);
+        }
+
+        [Fact]
+        public void Constructor_WithNullDataSource_ShouldHandleNullGracefully()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:24]: Same

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+            // Act
+            var item = new AmazonAthenaDataSourceItem(title, null);
+
+            // Assert
+            Assert.Equal(title, item.Title);
+        }
+
+        [Fact]
+        public void Constructor_WithEmptyTitle_ShouldHandleEmptyTitleGracefully()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:37]: Same

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
+using Reveal.Sdk.Dom.Data;
+using Xunit;
+
+namespace Reveal.Sdk.Dom.Tests.Data.DataSourceItems
+{
+    public class AmazonAthenaDataSourceItemFixture
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:6]: I am not sure we need unit tests for this kind of class. AmazonAthenaDataSourceItem does not change any fields / behaviour of its parent class. 
How do you think @hainv-ohio ?

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            // Act
+            var item = new AmazonAthenaDataSourceItem(title, dataSource);
+
+            // Assert
+            Assert.Equal(title, item.Title);
+            Assert.Equal(dataSource, item.DataSource);
+        }
+
+        [Fact]
+        public void HasTabularData_Should_Be_True()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:52]: Should be `Constructor_SetsHasTabularDataToTrue_WhenConstructed`

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            // Arrange
+            var item = new AmazonAthenaDataSourceItem("Test Item", new AmazonAthenaDataSource());
+
+            // Assert
+            Assert.True(item.HasTabularData);
+        }
+
+        [Fact]
+        public void HasAsset_Should_Be_False()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:62]: still incorrect name

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        {
+            // Arrange
+            var item = new AmazonAthenaDataSourceItem("Test Item", new AmazonAthenaDataSource());
+
+            // Assert
+            Assert.False(item.HasAsset);
+        }
+
+        [Fact]
+        public void Table_GetAndSetProperties_ReturnsCorrectValue()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:72]: change name

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs (hunk 5 above)
Comment (hainv-ohio) [reply src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:6]: It's true, as you said, but maybe only for now, in the future, that may change, so I think it's fine to keep the tests here. But @recca5p I don't think we need more tests than testing the happy case for the constructor. If we need to test the edge cases, we can test them in the parent class's tests.

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+            var actualTitle = item.Title;
+            var actualDataSource = item.DataSource;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+            Assert.Equal(dataSource, actualDataSource);
+        }
+
+        [Fact]
+        public void Constructor_SetsTitleAndDataSource_NullDataSource()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:26]: As discussed above, I think just remove the Null test cases, as this class only uses the parent class's constructor and doesn't do anything more.

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+            // Act
+            var actualTitle = item.Title;
+            var actualDataSource = item.DataSource;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+        }
+
+        [Fact]
+        public void Constructor_SetsTitle_EmptyTitle()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:42]: remove

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+            // Act
+            var actualTitle = item.Title;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+        }
+
+        [Fact]
+        public void HasTabularData_ReturnsTrue_Always()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:57]: Remove, only test new fields that this class add, if any, not the parent class's fields.

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+
+            // Act
+            var hasTabularData = item.HasTabularData;
+
+            // Assert
+            Assert.True(hasTabularData);
+        }
+
+        [Fact]
+        public void HasAsset_ReturnsFalse_Always()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:70]: same reason, remove

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+            // Act
+            var hasAsset = item.HasAsset;
+
+            // Assert
+            Assert.False(hasAsset);
+        }
+
+        [Fact]
+        public void Table_SetsAndGetsValue_ValidTableName()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonAthenaDataSourceItemFixture.cs:83]: remove

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonRedshiftDataSourceItemFixture.cs [hunk 14]
```diff
@@ -0,0 +1,69 @@
+using Reveal.Sdk.Dom.Data;
+using Xunit;
+
+namespace Reveal.Sdk.Dom.Tests.Data.DataSourceItems
+{
+    public class AmazonRedshiftDataSourceItemFixture
+    {
+        [Fact]
+        public void Constructor_SetsTitleAndDataSource_ValidDataSource()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonRedshiftDataSourceItemFixture.cs:9]: Only keep this test, remove the rest

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonRedshiftDataSourceItemFixture.cs [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            // Arrange
+            string title = "Test Item";
+            var dataSource = new AmazonRedshiftDataSource();
+            var item = new AmazonRedshiftDataSourceItem(title, dataSource);
+
+            // Act
+            var actualTitle = item.Title;
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonRedshiftDataSourceItemFixture.cs:17]: Move lines 13 14 in `// Arrange` to `// Act` section, and remove these lines

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+            var actualTitle = item.Title;
+            var actualDataSource = item.DataSource;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+            Assert.Equal(dataSource, actualDataSource);
+        }
+
+        [Fact]
+        public void Constructor_SetsTitleAndDataSource_NullDataSource()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs:27]: remove

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+            // Act
+            var actualTitle = item.Title;
+            var actualDataSource = item.DataSource;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+        }
+
+        [Fact]
+        public void Constructor_SetsTitle_EmptyTitle()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs:43]: remove

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+            var actualTitle = item.Title;
+            var actualDataSource = item.DataSource;
+
+            // Assert
+            Assert.Equal(title, actualTitle);
+            Assert.Equal(dataSource, actualDataSource);
+        }
+
+        [Fact]
+        public void HasAsset_ReturnsFalse_Always()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs:60]: remove

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+            // Arrange
+            var item = new AmazonS3DataSourceItem("Test Item", new AmazonS3DataSource());
+            var expectedPath = "s3://bucket-name/path/to/resource";
+
+            // Act
+            item.Path = expectedPath;
+            var actualPath = item.Path;
+
+            // Assert
+            Assert.Equal(expectedPath, actualPath);
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs:84]: Also check the `Properties` has the correct value set

File: src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+            var actualPath = item.Path;
+            var actualPropertyPath = item.Properties.GetValue<string>("Path");
+
+            // Assert
+            Assert.Null(actualPath);
+            Assert.Null(actualPropertyPath);
+        }
+
+        [Fact]
+        public void Properties_StoresPathValueCorrectly_ValidPath()
```
Comment (hainv-ohio) [inline src/Reveal.Sdk.Dom.Tests/Data/DataSourceItems/AmazonS3DataSourceItemFixture.cs:104]: remove, do it in the `Path_SetsAndGetsValue_ValidValue()` test

**Review by hainv-ohio — CHANGES_REQUESTED** [review]

**Review by hainv-ohio — APPROVED** [review]
